plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/metadata/MetadataRebuilder.java

```diff
@@ -238,23 +233,11 @@ public void end() {
      * Process exits from group level, executed in isolation.
      */
     private void rebuildMetadataExitGroup(final String currentGroupId) {
-      storageFacet.perform(db, new Operation<Void>()
```

Comment:
General rule of thumb is to mark methods as `@Transactional` at the lowest level you need the transaction, and only add `@Transactional` at the higher levels when there's additional cleanup/retry-logic required. So in this case we want to re-read the old metadata in `MetadataRebuilder` and re-merge our updates before retrying the put, rather than just retrying the put with the old data we'd merged before the conflict.